src/MainWindow.vala

```diff
@@ -180,154 +155,146 @@ public class Mail.MainWindow : Hdy.ApplicationWindow {
         refresh_stack = new Gtk.Stack () {
             transition_type = Gtk.StackTransitionType.CROSSFADE
         };
-        refresh_stack.add (refresh_button);
-        refresh_stack.add (refresh_spinner);
+        refresh_stack.add_child (refresh_button);
+        refresh_stack.add_child (refresh_spinner);
         refresh_stack.visible_child = refresh_button;
 
-        hide_read_switch = new Granite.SwitchModelButton (_("Hide read conversations"));
+       hide_read_switch = new Granite.SwitchModelButton (_("Hide read conversations"));
 
         hide_unstarred_switch = new Granite.SwitchModelButton (_("Hide unstarred conversations"));
 
-        var filter_menu_popover_grid = new Gtk.Grid () {
+        var filter_menu_popover_box = new Gtk.Box (Gtk.Orientation.VERTICAL, 0) {
             margin_bottom = 3,
-            margin_top = 3,
-            orientation = Gtk.Orientation.VERTICAL
+            margin_top = 3
         };
-        filter_menu_popover_grid.add (hide_read_switch);
-        filter_menu_popover_grid.add (hide_unstarred_switch);
-        filter_menu_popover_grid.show_all ();
+       filter_menu_popover_box.append (hide_read_switch);
+       filter_menu_popover_box.append (hide_unstarred_switch);
 
-        var filter_popover = new Gtk.Popover (null);
-        filter_popover.add (filter_menu_popover_grid);
+        var filter_popover = new Gtk.Popover ();
```

Comment:
Same link popover, you can remove the manual margins on the box if you add `Granite.STYLE_CLASS_MENU` here